### PR TITLE
Feature stop double queue 37

### LINKED ISSUE #37: stop double queue to the outpacketqueue on utcp_send
think I found another issue, when meshlink_send is called it just queues another packet to be sent and signals it, but where does it limit to just continue push packets to the queue when the upload is low?
this easily can cause tons of retransmits if not handled
https://github.com/meshlink/meshlink/blob/125279ed468788f67d4a6dc64353382bc88b8a6d/src/meshlink.c#L1268
https://github.com/meshlink/meshlink/blob/125279ed468788f67d4a6dc64353382bc88b8a6d/src/meshlink_queue.h#L46

### PR DESCRIPTION
**Pull request by everHannes**
Fixes #37 
based on #36 
requires https://github.com/meshlink/utcp/pull/18

### CODE REVIEW

**Comment by everHannes** [comment]
tested, works

File: src/meshlink.c [hunk 1]
```diff
@@ -2327,7 +2385,47 @@ static ssize_t channel_send(struct utcp *utcp, const void *data, size_t len) {
...
-    return meshlink_send(mesh, (meshlink_node_t *)n, data, len) ? len : -1;
+    // Prepare the packet
+    vpn_packet_t *packet = malloc(sizeof *packet);
+    if(!packet) {
+        meshlink_errno = MESHLINK_ENOMEM;
+        logger(mesh, MESHLINK_ERROR, "Error: channel_send packet memory allocation failed");
+        return false;
+    }
+
+    packet->probe = false;
```
Comment (a-krebs) [inline src/meshlink.c]: This is now duplicated code with `meshlink_send`. Can it be refactored into a common function that prepares the packet?

**Comment by a-krebs** [comment]
My only major concern is if there a fairness problem between `meshlink_send_from_queue` and utcp sending data directly. Is it possible that one could consistently fill the IO buffer on every loop, preventing the other from sending data?

**Comment by everHannes** [comment]
> My only major concern is if there a fairness problem between meshlink_send_from_queue and utcp sending data directly. Is it possible that one could consistently fill the IO buffer on every loop, preventing the other from sending data?

yes but unlikely since meshlink_send_from_queue takes precedence attempting to route just one packet while utcp send attempts to send for as much space there is remaining and is called last by the idle_cb, prior to the next select and a possible sleep (in code that would be upper but the sleep/select is what counts)

this looks reasonable to me since utcp send is buffered and has a retransmit mechanism while meshlink_send doesn't
plus we're not using meshlink_send that much if at all and it can be changed to a different socket otherwise

when this is found to be an issue the code could be overworked to improve shuffling of the calls

File: src/meshlink.c (hunk 1 above)
Comment (everHannes) [reply src/meshlink.c]: > This is now duplicated code with meshlink_send. Can it be refactored into a common function that prepares the packet?

done and further fixed the channel_send boolean returns

**Pull request merged**